client/actions/index.js

```diff
@@ -1,4 +1,23 @@
-import request from 'superagent'
+import axios from 'axios'
+
+const ROOT_URL = 'http://localhost:3000/posts'
+export const ADD_POST = 'ADD_POST'
+
+export function addNewPost (url) {
+  return dispatch => {
+    dispatch({type: 'ADD_NEW_POST_PENDING'})
+    return axios.post(`${ROOT_URL}`, { url })
+      .then(response => {
```

Comment:
This will probably have to change since `axios` is promise-based and `superagent` uses callbacks.